core/dagop.go

```diff
@@ -113,7 +113,7 @@ func (op FSDagOp) Exec(ctx context.Context, g bksession.Group, inputs []solver.R
 	switch inst := obj.(type) {
 	case dagql.Instance[*Directory]:
 		if inst.Self.Result != nil {
-			ref := worker.NewWorkerRefResult(inst.Self.Result, opt.Worker)
+			ref := worker.NewWorkerRefResult(inst.Self.Result.Clone(), opt.Worker)
```

Comment:
Ah neat, yup, I think this is what I was confused by - we need to `Clone` here, since otherwise we *would* be potentially double released refs.

LGTM :tada: